src/targets/gpu/device/sub.cpp

```diff
@@ -8,7 +8,7 @@ namespace device {
 
 void sub(hipStream_t stream, const argument& result, const argument& arg1, const argument& arg2)
 {
-    nary(stream, result, arg1, arg2)([](auto x, auto y) { return x - y; });
+    nary(stream, result, arg1, arg2)([] __device__ (auto x, auto y) { return x - y; });
```

Comment:
Why is `__device__` needed for this? It is only instantiated from a device function and is also `constexpr`.

Reply:
In CUDA/HIP syntax, kernel and device function can only call device function. By default this lambda is a host function, which cannot be called by a kernel or device function.

Reply:
But the lambda is `constexpr` so it should run on either host or device.

Reply:
Isn't there a flag we can enable with clang to make `constexpr` functions be `__device__ __host__` as default?